dlp/deid.py

```diff
@@ -0,0 +1,548 @@
+            'keyName' must also be set. Example:
+            contextFieldId = [{ 'name': 'user_id' }]
+        key_name: (Optional) The name of the Cloud KMS key used to encrypt
+            ('wrap') the AES-256 key. Example:
+            key_name = 'projects/YOUR_GCLOUD_PROJECT/locations/YOUR_LOCATION/
+            keyRings/YOUR_KEYRING_NAME/cryptoKeys/YOUR_KEY_NAME'
+        wrapped_key: (Optional) The encrypted ('wrapped') AES-256 key to use.
+            This key should be encrypted using the Cloud KMS key specified by
+            key_name.
+    Returns:
+        None; the response from the API is printed to the terminal.
+    """
+    # Import the client library
+    import google.cloud.dlp
+
+    # Instantiate a client
+    google.cloud.dlp.DlpServiceClient.SERVICE_ADDRESS = 'autopush-dlp.sandbox.googleapis.com' # DO NOT SUBMIT
+    dlp = google.cloud.dlp.DlpServiceClient()
+
+    # Add parent
+    parent = dlp.project_path(parent)
+
+    # Convert date field list to Protobuf type
+    def map_fields(field):
+        return {'name': field}
+
+    if date_fields:
+        date_fields = map(map_fields, date_fields)
+    else:
+        date_fields = []
+
+    # Read and parse the CSV file
+    import csv
+    from datetime import datetime
+    file = []
+    with open(input_csv_file, 'rb') as csvfile:
+        reader = csv.reader(csvfile)
+        for row in reader:
+            file.append(row)
+
+    #  Helper function for converting CSV rows to Protobuf types
+    def map_headers(header):
+        return {'name': header}
+
+    def map_data(value):
+        try:
+            date = datetime.strptime(value, '%m/%d/%Y')
+            return {
+                'date_value': {
+                    'year': date.year,
+                    'month': date.month,
+                    'day': date.day
+                }
+            }
+        except ValueError:
+            return {'string_value': value}
+
+    def map_rows(row):
+        return {'values': map(map_data, row)}
+
+    csv_headers = map(map_headers, file[0])
+    csv_rows = map(map_rows, file[1:])
+
+    # Construct the table dict
+    table_item = {
+        'table': {
+            'headers': csv_headers,
+            'rows': csv_rows
+        }
+    }
+
+    # Construct date shift config
+    date_shift_config = {
+        'lower_bound_days': lower_bound_days,
+        'upper_bound_days': upper_bound_days
+    }
+
+    # If using Cloud KMS key
+    if context_field_id and key_name and wrapped_key:
+        import base64
+        date_shift_config['context'] = {'name': context_field_id}
+        date_shift_config['crypto_key'] = {
+            'kms_wrapped': {
+                'wrapped_key': base64.b64decode(wrapped_key),
+                'crypto_key_name': key_name
+            }
+        }
+    elif context_field_id or key_name or wrapped_key:
+        raise StandardError("""You must set either ALL or NONE of
```

Comment:
Change this to `raise ValueError(...)`. ValueError is preferred when an issue exists with function arguments or other incorrect data.

Reply:
done